bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/classpath/ClasspathHolder.java

```diff
@@ -0,0 +1,72 @@
+/*-
+ * Copyright (c) 2024 Salesforce and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Salesforce - adapted from M2E, JDT or other Eclipse project
+ */
+package com.salesforce.bazel.eclipse.core.classpath;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IPath;
+
+import com.salesforce.bazel.eclipse.core.model.discovery.classpath.ClasspathEntry;
+
+/**
+ * Represents a targets classpath as two parts, loaded being the classpath entries that are loaded into the project
+ * model, and unloaded being classpaths that are part of the target, but are not to be loaded in the project model.
+ *
+ * Note: Unloaded contains null values to keep the order between loaded and unloaded
+ */
+public record ClasspathHolder(Collection<ClasspathEntry> loaded, Collection<ClasspathEntry> unloaded) {
```

Comment:
Can we rename to `compile` and `transitive` dependencies? I think that better communicates the design/intent.